core/kazoo_provisioner/src/provisioner_v5.erl

```diff
@@ -267,11 +265,20 @@ settings_lines(Device) ->
 
 -spec settings_basic(kzd_devices:doc()) -> kz_json:object().
 settings_basic(Device) ->
-    Enabled = case kzd_devices:enabled(Device, 'undefined') of
-                  'undefined' -> 'undefined';
-                  Else -> kz_term:is_true(Else)
-              end,
-    Props = [{<<"display_name">>, kzd_devices:name(Device)}
+    AccountId = kz_doc:account_id(Device),
+    {'ok', Account} = kzd_accounts:fetch(AccountId),
+    {'ok', Owner} = case kzd_devices:owner_id(Device) of
+                        'undefined' -> {'ok', kz_json:new()};
+                        OwnerId -> kzd_users:fetch(AccountId, OwnerId)
+                    end,
+    Enabled = kzd_devices:enabled(Device)
+        andalso kzd_accounts:is_enabled(Account)
+        andalso kzd_users:enabled(Owner),
+    DisplayName = case kzd_accounts:name(Account) of
```

Comment:
I think you want to take the user's name into account. I would maybe create a list of possible names, `[NameFromUser, NameFromDevice, NameFromAccount]` for display name.

Reply:
Then go until you get a non-empty name. See `kz_endpoint:merge_attribute_caller_id/4` for an idea.